### DISCUSSION TITLE
Avalonia LSP in Neovim

### DISCUSSION THREAD

**Discussion by mrphil2105** [AvaloniaUI/Avalonia#15322] [Q&A]
Hey! I am trying to get the Avalonia LSP for VSCode to work in Neovim. It seems to be active and somewhat working. But it tells me to build the project to get completion when I call the built-in LSP completion function `vim.lsp.buf.completion`. But the project is already built. Any idea what might be causing this, and what more information do you need to help me fix it?

Here's how I add the LSP to my config:
```lua
local configs = require("lspconfig.configs")
local root_pattern = require("lspconfig.util").root_pattern
if not configs.avalonia then
    configs.avalonia = {
        default_config = {
            cmd = { "dotnet", "/path/to/AvaloniaLanguageServer.dll" },
            root_dir = root_pattern("*.csproj"),
            filetypes = { "axaml", "xml" },
        },
    }
end

require("lspconfig").avalonia.setup(coq.lsp_ensure_capabilities {
    on_attach = on_attach,
    settings = {},
})
```

Thanks in advance.

**Comment by timunie** [comment]
@mrphil2105 I feel like this is a more complex topic and may need paid support to handle properly (if even supported). So I suggest to reach out to the team if you are interested in that.  https://www.avaloniaui.net/support

**Reply by mrphil2105** [reply]
I am only a student and cannot afford this kind of support. I think the issue is the way it checks for the binary dll file. Maybe I have set it up wrong, but I am not sure.

**Reply by timunie** [reply]
Understood 👍

**Reply by kekekeks** [reply]
LSP sources are here - https://github.com/AvaloniaUI/AvaloniaVSCode/tree/main/src/AvaloniaLSP

**Reply by mrphil2105** [reply]
I tried looking at it, but I cannot figure out what leads it to say that I should build the project. I am not sure if it even gets to the point of looking for the DLL file.

**Reply by ADIX7** [reply]
You should also set the filetype to xml. You can manually set it with `:set ft=xml` or add this to your lua config:

```lua
-- set filetype xml for extension .axaml
vim.api.nvim_create_autocmd('BufRead', {
    desc = 'set filetype xml for extension .axaml',
    pattern = '*.axaml',
    callback = function()
        vim.bo.filetype = 'xml'
    end,
})

```

**Comment by SaverinOnRails** [comment]
In case anyone still needs a solution to this, i've managed to port the LSP to standalone editors like vim, neovim, e.t.c

https://www.github.com/eugenenoble2005/avalonia-ls.git

**Reply by kekekeks** [reply]
You can run previewer using `dotnet msbuild /t:AvaloniaFilePreview /p:APreviewFile=MyWindow.xaml`, it will resolve stuff automatically.

See https://github.com/AvaloniaUI/Avalonia/pull/12206

**Reply by SaverinOnRails** [reply]
Yes I've also added a helper for this to the project above

```
avalonia-preview ---file window.axaml ---port 8080
```
To preview in the browser.

I'm also making research into rendering the preview in the terminal via kitty image protocol.
